### PR TITLE
Transformation Lifecycle architecture

### PR DESCRIPTION
**Pull request by c-mueller**
The included document currently represents more of a note document it is incomplete

### CODE REVIEW

File: docs/dev/plugins/transformation-lifecycle/lifecycle-architecture.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# Transformation Lifecycle Archtecture
```
Comment (nfode) [inline docs/dev/plugins/transformation-lifecycle/lifecycle-architecture.md]: `Archtecture` -> `Architecture`

File: docs/dev/plugins/transformation-lifecycle/lifecycle-architecture.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+## Introduction
+
+This document and the corresponding diagrams represent a potential architecture for the lifecycle operations of a transformation described in Pull Request #159. It was expected, that a transformation has to support the following operations (descriptions have been taken from [here](transformation-lifecycle.md)):
```
Comment (nfode) [inline docs/dev/plugins/transformation-lifecycle/lifecycle-architecture.md]: Might be good to add a link to the PR.

File: docs/dev/plugins/transformation-lifecycle/lifecycle-architecture.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    void validateModel();
+    void prepare();
+    void transform();
+    void cleanup();
+}
+```
+
+#### Combined model and type check
+
+## Progress caluclation
```
Comment (nfode) [inline docs/dev/plugins/transformation-lifecycle/lifecycle-architecture.md]: `caluclation` -> `calculation`

File: docs/dev/plugins/transformation-lifecycle/lifecycle-architecture.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+4. `transform()` - this is where the real transformation is happening.
+5. `clean()` - in this phase the plugin cleans up remaining leftovers from the previos steps. This can be for example files generated during the transformation that are not part of the target artifact.
+
+The lifecyle approach is resolving several general issues:
+
+- Splitting the transformation into tasks (lifecycle operations) allows a very simple progress calculation. (More information will follow below)
+- Simplifies the plugin implementation in terms of code size because common parts can get externalized (removes boilerplate code)
+- The plugins structure gets way clearer.
+
+The TransformationLifecycleInterface (short: TLI) has methods that represent each task these methods get called in the sequence that was defined in PR #159 (or above).
```
Comment (nfode) [inline docs/dev/plugins/transformation-lifecycle/lifecycle-architecture.md]: Again: Link to the PR

File: docs/dev/plugins/transformation-lifecycle/lifecycle-architecture.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+This approach clearly splits the node type check (short `type check`) and the model check into two seperate phases. 
+
+The interfaces defined by using this approach will look like this (just to showcase):
+
+TOSCAna Plugin (just for illustration purposes, class diagrams will follow):
+```java
+public abstract class LifecycleAwarePlugin<T extends TransformationLifecycle>
+    implements TransformationPlugin {
+    protected abstract List<NodeTypes> getSupportedNodeTypes();
```
Comment (hnicke) [inline docs/dev/plugins/transformation-lifecycle/lifecycle-architecture.md]: Should be a Set, not a List. It might be Set<Class>.

**Comment by nfode** [comment]
@c-mueller what is with the .wsd files? We are still using LucidChart so please remove them

**Comment by c-mueller** [comment]
The `.wsd` file represents the UML Sequence markup that is used by lucid charts to render the sequence diagram. there is no plant uml envolved.

I Honestly dont like the idea of removing the file as s source in that case

**Comment by c-mueller** [comment]
Due to the recent meetings and the discussion about the `wsd` file i consider this Pull request as `approved` and i will therefore merge it.

**Pull request merged**